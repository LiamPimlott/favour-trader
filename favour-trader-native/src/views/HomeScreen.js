import React, { Component } from 'react';
import { FlatList, StyleSheet, View, Picker, Text, TextInput, TouchableOpacity } from 'react-native';
import { StackNavigator } from 'react-navigation';
import {Divider} from 'react-native-elements'
import AuthService from "../components/AuthService";
import axios from 'axios';
import MatchCard from "../components/MatchCard";

export default class HomeScreen extends React.Component {
    constructor() {
        super();
        this.state = {
            matches: '',
            matchedUsers: [],
        };
        this.authService = new AuthService();
        this.updateMatches = this.updateMatches.bind(this);
    }

    componentDidMount() {
        this.mounted = true;
    }

    componentWillUnmount() {
        this.mounted = false;
    }

    updateMatches = (matchFilter) => {
        if (this.authService.loggedIn() && this.mounted && matchFilter !== 'Filter the Matches') {
            this.setState({matches: matchFilter});
            const config = {
                headers: {
                    Authorization: ''
                },
                params: {
                    has: '',
                    wants: ''
                }
            };
            this.authService.getToken().then((token) => {
                config.headers.Authorization = token;
                if (this.state.matches === 'has') {
                    config.params = {
                        hasFilter: 'true',
                        wantsFilter: 'false'
                    };
                } else if (this.state.matches === 'wants') {
                    config.params = {
                        hasFilter: 'false',
                        wantsFilter: 'true'
                    };
                } else if (this.state.matches === 'perfect') {
                    config.params.has = 'true';
                    config.params.wants = 'true';
                }
                axios.get('http://favour-trader.appspot.com/api/users/matches', config)
                    .then(res => res.data.matches)
                    .then(matches => {
                        this.setState({matchedUsers: matches});
                    })
                    .catch((err) => {
                        console.log(err);
                    });
            });
        }

    };

    _renderItem = ({item}) => (
        <MatchCard match={item}/>
    );
    _keyExtractor = (item, index) => item._id;

    render() {
        return (
            <View style={styles.container}>
                <Picker selectedValue={this.state.matches}
                        onValueChange={this.updateMatches}
                        style={styles.picker}>
                    <Picker.Item label="Filter the Matches" value="0"/>
                    <Picker.Item label="What I Want" value="wants"/>
                    <Picker.Item label="What I Have" value="has"/>
                    <Picker.Item label="Perfect Matches" value="perfect"/>
                </Picker>
                <Divider style={styles.divider} />
                <FlatList
                    data={this.state.matchedUsers}
                    keyExtractor={this._keyExtractor}
                    renderItem={this._renderItem}
                />
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        alignItems: 'center',
    },
    picker: {
        width: 300,
    },
    divider: {
        height: 1,
        width: 300,
        backgroundColor: 'grey'
    }
});
