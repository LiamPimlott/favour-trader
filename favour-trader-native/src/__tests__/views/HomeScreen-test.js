import 'react-native';
import React from 'react';
import { shallow, configure } from 'enzyme';
import HomeScreen from '../../views/HomeScreen.js';
import toJson from 'enzyme-to-json'; //added this line

const Adapter = require("enzyme-adapter-react-16");
configure({adapter : new Adapter()});

describe('Testing HomeScreen component', () => {
    it('renders as expected', () => {
        const wrapper = shallow(
            <HomeScreen/>
        );
        expect(toJson(wrapper)).toMatchSnapshot(); //edited this line
    });
});