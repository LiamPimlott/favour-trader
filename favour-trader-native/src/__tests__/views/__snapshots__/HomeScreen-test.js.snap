// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing HomeScreen component renders as expected 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
    }
  }
>
  <Button
    TouchableComponent={[Function]}
    buttonStyle={
      Object {
        "borderRadius": 0,
        "marginBottom": 10,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 10,
        "width": 300,
      }
    }
    clear={false}
    disabled={false}
    iconRight={false}
    loadingProps={
      Object {
        "color": "white",
        "size": "small",
      }
    }
    onPress={[Function]}
    title="Select Filter"
  />
  <View />
  <Divider
    style={
      Object {
        "backgroundColor": "grey",
        "height": 1,
        "width": 300,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    No matches found or Update selection from picker!
  </Text>
</View>
`;

exports[`Testing Login component renders as expected 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
    }
  }
>
  <Button
    TouchableComponent={[Function]}
    buttonStyle={
      Object {
        "borderRadius": 0,
        "marginBottom": 10,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 10,
        "width": 300,
      }
    }
    clear={false}
    disabled={false}
    iconRight={false}
    loadingProps={
      Object {
        "color": "white",
        "size": "small",
      }
    }
    onPress={[Function]}
    title="Select Filter"
  />
  <View />
  <Divider
    style={
      Object {
        "backgroundColor": "grey",
        "height": 1,
        "width": 300,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    No matches found or Update selection from picker!
  </Text>
</View>
`;
